### DISCUSSION TITLE
AZ Touch + ESP32-WROOM-32

### DISCUSSION THREAD

**Discussion by MatijaTerzic** [HASwitchPlate/openHASP#131] [Hardware]
I have the following: https://www.az-delivery.de/nl/products/az-touch-wandgehauseset-mit-touchscreen-fur-esp8266-und-esp32

In arduino IDE i use #include <SPI.h> and #include <TFT_eSPI.h>

I have modified User_Setup.h of eSPI library to the following:

```c
#define TFT_BL   15            // LED back-light control pin
#define TFT_BACKLIGHT_ON LOW  // Level to turn ON back-light (HIGH or LOW)
#define TFT_MISO 19
#define TFT_MOSI 23
#define TFT_SCLK 18
#define TFT_CS    5  // Chip select control pin
#define TFT_DC    4  // Data Command control pin
#define TFT_RST  22  // Reset pin (could connect to RST pin)
//#define TFT_RST  -1  // Set TFT_RST to -1 if display RESET is connected to ESP32 board RST

#define TOUCH_CS 14     // Chip select pin (T_CS) of touch screen
```

I would like to custom build / compile openhasp to work like this. What do i need to modify for this ? I have tried the same User_Setup.h in openhasp 

Also have the following in the ini file:

```ini
[env:esp32dev-ili9488]
extends = esp32
board = esp32dev

build_flags =
    ${env.build_flags}
    ${esp32.build_flags}

;region -- TFT_eSPI build options ------------------------
    -D ILI9341_DRIVER=1
    -D TFT_WIDTH=240
    -D TFT_HEIGHT=320
    -D TFT_ROTATION=0 ; see TFT_ROTATION values
    -D INVERT_COLORS=1  ; to fix colors
    -D SPI_FREQUENCY=60000000
    -D SPI_TOUCH_FREQUENCY=2500000
    -D SPI_READ_FREQUENCY=20000000
    -D USER_SETUP_LOADED=1
    -D TOUCH_DRIVER=2046     ; XPT2046 Resistive SPI touch panel driver
    -D SUPPORT_TRANSACTIONS
    ${esp32.vspi}        ; Use VSPI hardware SPI bus: 
    -D TFT_MISO=19
    -D TFT_MOSI=23
    -D TFT_SCLK=18
    ;-D MISO    =8
    ;-D MOSI    =4
    ;-D CLK     = 3
; wiring recommendations, change pins according to your wiring
    -D TFT_DC=4          ; DC,  lcd pin 3
    ;-D TFT_RST=15        ; RES, lcd pin 5
    ;-D TFT_BCKL=15       ; BLK, lcd pin 7 (configurable via web UI (e.g. 21))
    -D TFT_CS=5         ; CS1, lcd pin 9
    -D TOUCH_CS=14       ; CS2, lcd pin 10  
;endregion

; -- Debugging options -----------------------------
;   -D CORE_DEBUG_LEVEL=ARDUHAL_LOG_LEVEL_DEBUG

;region -- Library options -------------------------------
lib_deps =
    ${env.lib_deps}
    ${esp32.lib_deps}

lib_ignore =
    ${env.lib_ignore}
    ${esp32.lib_ignore}
;endregion
```

But the display does nothing (webinterface works). In the log / terminal i see a line that the backlight pin is not configured and has to be done in webinterface but i can't set it to PIN 15 / inverted (LOW)

Is their a solution ?

**Comment by nagyrobi** [comment]
What a great find!!
Which esp module card do you use with it?

**Reply by MatijaTerzic** [reply]
I use the ESP32-WROOM-32 with it. Also compatible with D1 Mini.

I love it. Just perfect in combination with Home Assistant !

**Comment by fvanroie** [comment]
The whole tft_espi configuration is done via the ini file. The backlight can also be set in the web interface.
Pin 15 is not availabe, but you can set it also via the command-line: `config/gui {"bckl":15}` and `reboot` to save.

If that works, we'll need to add the pin as a valid option in the web UI too.
Inverted backlight low is not yet supported yet, but you could use `dim=0` to set it to full brightness...

**Comment by arovak** [comment]
could you test following config in a .ini file in user_setups/active ?

```ini
[env:arduitouch-esp32_ili9341]
extends = esp32
board = esp32dev

build_flags =
    ${env.build_flags}
    ${esp32.build_flags}

;region -- TFT_eSPI build options ------------------------
    ${lcd.lolin24}
    ${esp32.vspi}        ; Use VSPI hardware SPI bus
    -D TFT_DC=4
    -D TFT_CS=5 
    -D TFT_RST=22        ; RST
    -D TFT_BCKL=15
    -D TOUCH_CS=14
;endregion

;region -- Library options -------------------------------
lib_deps =
    ${env.lib_deps}
    ${esp32.lib_deps}

lib_ignore =
    ${env.lib_ignore}
    ${esp32.lib_ignore}
;endregion
```

**Reply by MatijaTerzic** [reply]
Hello,

This indeed works but i have to also do a dim=0 in cmdline

Thanks

**Reply by MatijaTerzic** [reply]
correction, i need to do "config/gui {"bckl":15}" and reboot. How do i make pin 15 available in the Web UI ? @fvanroie

**Reply by fvanroie** [reply]
> This indeed works but i have to also do a dim=0 in cmdline

If that `arduitouch-esp32_ili9341` from @arovak works, we can add it to the known user_setups.
Inverted (low) dimming is a new feature that needs to be added, so for the time being you need to invert the number yourself.

> I need to do "config/gui {"bckl":15}" and reboot.

`TFT_BCKL` is only taken into account when the device is blank. After the config is written once, the `config/gui` value is used.

> How do i make pin 15 available in the Web UI ? @fvanroie

It's in hasp_http.cpp, I have added the pin in the master branch.

**Reply by MatijaTerzic** [reply]
> > This indeed works but i have to also do a dim=0 in cmdline
> 
> If that `arduitouch-esp32_ili9341` from @arovak works, we can add it to the known user_setups.
> Inverted (low) dimming is a new feature that needs to be added, so for the time being you need to invert the number yourself.
> 
Yeah it surely works. Beeing in the proces of creating pages allready ;). 
How do i do that ?

> > I need to do "config/gui {"bckl":15}" and reboot.
> 
> `TFT_BCKL` is only taken into account when the device is blank. After the config is written once, the `config/gui` value is used.
> 
Oke i see.

> > How do i make pin 15 available in the Web UI ? @fvanroie
> 
> It's in hasp_http.cpp, I have added the pin in the master branch.
Thanks !!

I am very happy with the speed that this ticket is picked-up and answered !!

**Reply by fvanroie** [reply]
>  How do i do that ?

You can do it in several ways to [create pages](https://haswitchplate.github.io/openHASP-docs/#pages/):

- [Command line](https://haswitchplate.github.io/openHASP-docs/#commands/#jsonl) to create objects: `jsonl {"obj":"btn","id":1,"x":50,"y":50}`
- Via mqtt `command/jsonl` and payload (max 2kb)
- [Upload](https://haswitchplate.github.io/openHASP-docs/#configuration/hasp/) an [example](https://haswitchplate.github.io/openHASP-docs/#example-2/) [pages.jsonl](https://haswitchplate.github.io/openHASP-docs/#example-0/) file to the local filesystem

**Reply by MatijaTerzic** [reply]
@fvanroie I think you misunderstood me :). Creating pages.jsonl is not a problem but what i mean is how do i invert the number (low dimming) ?

**Reply by arovak** [reply]
> Yeah it surely works. Beeing in the proces of creating pages allready ;).

Thanks for confirming we will add this setup to the known devices list!

At this moment you need to have your home automation software handle this. Send a `dim 0` command to turn the LCD to full brightness or `dim 100` for 0 brightness. 
The dim command needs to be send on the MQTT topic `hasp/_platename_/command/dim`

**Reply by fvanroie** [reply]
For the initial brightness at power-on, there is the [`startdim`](https://haswitchplate.github.io/openHASP-docs/#commands/#configsubmodule) parameter in the webUI or console:

```
config/hasp {"startdim":0}
reboot
```

When using templates etc. you need to calculate 255 - {{ dimlevel }}.
It's annoying, but we'll see what we can do to add this feature.

**Reply by MatijaTerzic** [reply]
@arovak The MQTT command works like a charm. But its 0 to 255 ;)

**Reply by MatijaTerzic** [reply]
Also setup the custom integration for Home Assistant and it is just great. Love it all the way !

**Reply by fvanroie** [reply]
The dim-level has been changed just recently from 0-100% to 0-255 which is more in line with the other gpio's.
So it's dependent on which firmware version you're running.

**Reply by HA-TB303** [reply]
> Thanks for confirming we will add this setup to the known devices list!

Any chance of that happening soon? I have one of these in he exact same setup but can't get it to work.

**Reply by fvanroie** [reply]
Which bin file and ESP are you using?

I have just now added `arduitouch-esp32_ili9341_full_4MB_v0.6.2.bin` to the supported builds and tested it on the ArduiTouch 2.4" with an ESP32 DevKit C (38-pin). To download, click on the [latest nightly build](https://github.com/HASwitchPlate/openHASP/actions) and there you can download `openHASP firmware.zip` with the firmware binary.

Let me know if you still have issues with that build.

**Reply by fvanroie** [reply]
@HA-TB303 Did you manage to get it to work with the nightly build?

**Comment by HA-TB303** [comment]
No, I did not.

**Reply by fvanroie** [reply]
Can you elaborate on which binary file you are using and what it is exactly that is not working for you?
I also have one of these device now and just tested it with the latest build without issues.

If I know what error or behavior you are experiencing, I'm able to help out...

**Comment by HA-TB303** [comment]
Hi, I'm using: arduitouch-esp32_ili9341_full_4MB_v0.6.2.bin, after booting the display lights up, but remains empty.
I can configure the module (wifi) and made a screenshot:
At that point the display should contain:
![image](https://user-images.githubusercontent.com/9609805/136627606-942099cf-0509-4a48-8f56-9366b9d6c4f7.png)
but it looks like:
![image](https://user-images.githubusercontent.com/9609805/136627727-ece58b8a-9baf-4a5f-978a-8f8b845a5a25.png)

Thanks,

Michel

**Reply by fvanroie** [reply]
Thank you for the pictures, that really helps! The screenshot indicates that the firmware and graphics engine is functioning in the background.

A white screen is typical of a problem with the pin assignment (software config) or wiring (hardware):

## Software config

When the ESP32 boots up, check the serial log to find the TFT section. It should be:
```
[22:19:33.362][113792/198888 42][    0/    0  0] TFT : Starting...
[22:19:33.670][113792/198756 42][    0/    0  0] TFT : Rotation   : 0
[22:19:33.682][113792/198756 42][    0/    0  0] TFT : Invert Disp: No
[22:19:33.728][113792/198756 42][    0/    0  0] TFT : TFT_eSPI   : v2.3.70
[22:19:33.740][113792/198756 42][    0/    0  0] TFT : Transactns : Yes
[22:19:33.752][113792/198756 42][    0/    0  0] TFT : Interface  : SPI
[22:19:33.764][113792/198756 42][    0/    0  0] TFT : Driver     : ILI9341
[22:19:33.777][113792/198756 42][    0/    0  0] TFT : Resolution : 240x320
[22:19:33.789][113792/198756 42][    0/    0  0] TFT : MOSI       : 23 (GPIO 23)
[22:19:33.802][113792/198756 42][    0/    0  0] TFT : MISO       : 19 (GPIO 19)
[22:19:33.815][113792/198756 42][    0/    0  0] TFT : SCLK       : 18 (GPIO 18)
[22:19:33.828][113792/198756 42][    0/    0  0] TFT : TFT_CS     : 5 (GPIO 05)
[22:19:33.841][113792/198756 42][    0/    0  0] TFT : TFT_DC     : 4 (GPIO 04)
[22:19:33.854][113792/198756 42][    0/    0  0] TFT : TFT_RST    : 22 (GPIO 22)
[22:19:33.867][113792/198756 42][    0/    0  0] TFT : TOUCH_CS   : 14 (GPIO 14)
[22:19:33.880][113792/198756 42][    0/    0  0] TFT : Display SPI freq. : 40.0 MHz
[22:19:33.893][113792/198756 42][    0/    0  0] TFT : Touch SPI freq.   : 2.5 MHz
[22:19:33.906][113792/198756 42][    0/    0  0] TFT : Started
```
## Hardware

If the pin configuration is OK, the next step is checking the connection from the ESP to the display:
- Can you check the solder connections for any shorts?
- Test with a multi-meter if there is a good connection from the ESP pin headers to the display pinheader
- Could you try to hook up the display with jumper wires instead? Just the display is enough, leave touch unconnected:

Number | Pin Label | Description
-- | -- | --
1 | VCC | 5V/3.3V power input
2 | GND | Ground
3 | CS | GPIO 5
4 | RESET | GPIO 22
5 | DC | GPIO 4
6 | SDI(MOSI) | GPIO 23
7 | SCK | GPIO 18
8 | LED | GPIO 15
9 | SDO(MISO) | GPIO 19

Lastly, which ESP32 dev board are you using? This is my setup:
![image](https://user-images.githubusercontent.com/15969459/136632091-1819cadb-3713-4a0b-8ee6-7baa790d0d7e.png)

**Reply by HA-TB303** [reply]
Hi, I was not able to test any sooner. I was about to give up and then I removed the esp32-devkit and placed it again, and now it works. It must have been a faulty contact.

Thanks for all your help!

**Comment by Thomaskru** [comment]
Hi!
arduitouch-esp32_ili9341_full_4MB_v0.6.2.bin worked out of the box on Ardui-Touch Kit with 2,8" Display and ESP32 DevKitc_v4.

HomeAssistant integration also works but I am failing with a custom image for DTH22 Sensor used on GPIO32.

**Reply by fvanroie** [reply]
What steps did you take to compile the custom image? What error or result did you get exactly?

**Reply by Thomaskru** [reply]
Hi!
Steps I have done:
Git clone HASP repo
enable "#define HASP_USE_CUSTOM 1" in user_config_override.h
Add my_custom.h file (content below)
Firmware build failed because of missing libraries. So i have added missing libraries in Visual Code/Plattform.io 
After this steps (mostly trial and error) firmware build was ok, but i have no clue what to do next nor if these steps are correct. (Sorry im newbie in coding) Maybe all after void setup is wrong here?

```c
/* MIT License - Copyright (c) 2019-2021 Francis Van Roie
   For full license information read the LICENSE file in the project folder */

// USAGE: - Copy this file and rename it to my_custom.h
//        - uncomment in your user_config_override.h the line containing #define HASP_USE_CUSTOM 1
//

#ifndef HASP_CUSTOM_H
#define HASP_CUSTOM_H

// DHT Temperature & Humidity Sensor
// Unified Sensor Library Example
// Written by Tony DiCola for Adafruit Industries
// Released under an MIT license.

// REQUIRES the following Arduino libraries:
// - DHT Sensor Library: https://github.com/adafruit/DHT-sensor-library
// - Adafruit Unified Sensor Lib: https://github.com/adafruit/Adafruit_Sensor

#include <Adafruit_Sensor.h>
#if defined(HASP_USE_CUSTOM)
#include <DHT.h>
#if defined(HASP_USE_CUSTOM)
#include <DHT.h>
#if defined(HASP_USE_CUSTOM)

//#define DHTPIN 2     // Digital pin connected to the DHT sensor
#define DHTPIN 32 // Digital pin connected to the DHT sensor
// Feather HUZZAH ESP8266 note: use pins 3, 4, 5, 12, 13 or 14 --
// Pin 15 can work but DHT must be disconnected during program upload.

// Uncomment the type of sensor in use:
//#define DHTTYPE    DHT11     // DHT 11
#define DHTTYPE DHT22 // DHT 22 (AM2302)
//#define DHTTYPE    DHT21     // DHT 21 (AM2301)

// See guide for details on sensor wiring and usage:
//   https://learn.adafruit.com/dht/overview

DHT_Unified dht(DHTPIN, DHTTYPE);

uint32_t delayMS;

void setup()
{
    Serial.begin(9600);
    // Initialize device.
    dht.begin();
    Serial.println(F("DHTxx Unified Sensor Example"));
    // Print temperature sensor details.
    sensor_t sensor;
    dht.temperature().getSensor(&sensor);
    Serial.println(F("------------------------------------"));
    Serial.println(F("Temperature Sensor"));
    Serial.print(F("Sensor Type: "));
    Serial.println(sensor.name);
    Serial.print(F("Driver Ver:  "));
    Serial.println(sensor.version);
    Serial.print(F("Unique ID:   "));
    Serial.println(sensor.sensor_id);
    Serial.print(F("Max Value:   "));
    Serial.print(sensor.max_value);
    Serial.println(F("°C"));
    Serial.print(F("Min Value:   "));
    Serial.print(sensor.min_value);
    Serial.println(F("°C"));
    Serial.print(F("Resolution:  "));
    Serial.print(sensor.resolution);
    Serial.println(F("°C"));
    Serial.println(F("------------------------------------"));
    // Print humidity sensor details.
    dht.humidity().getSensor(&sensor);
    Serial.println(F("Humidity Sensor"));
    Serial.print(F("Sensor Type: "));
    Serial.println(sensor.name);
    Serial.print(F("Driver Ver:  "));
    Serial.println(sensor.version);
    Serial.print(F("Unique ID:   "));
    Serial.println(sensor.sensor_id);
    Serial.print(F("Max Value:   "));
    Serial.print(sensor.max_value);
    Serial.println(F("%"));
    Serial.print(F("Min Value:   "));
    Serial.print(sensor.min_value);
    Serial.println(F("%"));
    Serial.print(F("Resolution:  "));
    Serial.print(sensor.resolution);
    Serial.println(F("%"));
    Serial.println(F("------------------------------------"));
    // Set delay between sensor readings based on sensor details.
    delayMS = sensor.min_delay / 1000;
}

void loop()
{
    // Delay between measurements.
    delay(delayMS);
    // Get temperature event and print its value.
    sensors_event_t event;
    dht.temperature().getEvent(&event);
    if(isnan(event.temperature)) {
        Serial.println(F("Error reading temperature!"));
    } else {
        Serial.print(F("Temperature: "));
        Serial.print(event.temperature);
        Serial.println(F("°C"));
    }
    // Get humidity event and print its value.
    dht.humidity().getEvent(&event);
    if(isnan(event.relative_humidity)) {
        Serial.println(F("Error reading humidity!"));
    } else {
        Serial.print(F("Humidity: "));
        Serial.print(event.relative_humidity);
        Serial.println(F("%"));
    }
}

#endif // HASP_USE_CUSTOM

#endif // HASP_CUSTOM_H

/* MIT License - Copyright (c) 2019-2021 Francis Van Roie
   For full license information read the LICENSE file in the project folder */

// USAGE: - Copy this file and rename it to my_custom.h
//        - uncomment in your user_config_override.h the line containing #define HASP_USE_CUSTOM 1
//
```

**Reply by derjoerg** [reply]
Hi,
Can you share your ini file?
I also have the az touch mod with 2.8 display and the esp32 dev.
The weatherstation example worked out of the box, so I think I soldered everything correctly.

I flashed the same image as you.

The web-interface is working, but the display keeps dark :(

Joerg

**Reply by fvanroie** [reply]
@derjoerg Which bin file did you upload to the ESP32 DevKitc_v4? Do you have a link to your esp32dev, so we can verify the pinout?
Be sure to configure the backlight pin in Configuration > Display. Please also post the bootlog showing the TFT pin configuration.

**Reply by derjoerg** [reply]
Thank you, thank you, thank you!

I just needed to define GPIO15 as backlight pin and it worked :)

**Comment by fvanroie** [comment]
A [dedicated page for the AZ-Touch](https://haswitchplate.github.io/openHASP-docs/latest/devices/az-touch/) has been added to the documentation.

**Comment by oh5lep** [comment]
I have similar setup: AZ-Touch Mod and ESP32 Dev KitC V4 WROOM-32. Firmware loads ok, I can get IP, but after closing IP dialog by clicking OK, there is only text 'plate' in upper left corner. 
Should it show something more by default?

Log:
---
```
rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:12596
load:0x40080400,len:2916
entry 0x400805c4



        open____ _____ _____ _____
          |  |  |  _  |   __|  _  |
          |     |     |__   |   __|
          |__|__|__|__|_____|__|
        Home Automation Switch Plate
        Open Hardware edition v0.6.3

[       0.080][110580/169492 34][    0/    0  0] DBUG: Started @ 115200 bps
#[       0.091][110580/169492 34][    0/    0  0] DBUG: Environment: az-touch-mod-esp32_ili9341_4MB
#[       0.105][110580/169492 34][    0/    0  0] UART: Started
#[       0.116][110580/169492 34][    0/    0  0] UART: Client login from serial
#[       0.128][110580/169492 34][    0/    0  0] CONF: SPI flash FS mounted
Prompt > [       0.160][110580/169492 34][    0/    0  0] CONF: Loading /config.json
Prompt > [       0.174][110580/168872 34][    0/    0  0] CONF: Loaded /config.json
Prompt > [       0.186][110580/169552 34][    0/    0  0] DBUG: Loading debug settings
Prompt > [       0.200][110580/169552 34][    0/    0  0] GPIO: Loading GUI settings
Prompt > [       0.213][110580/169552 34][    0/    0  0] GUI : First Touch Calibration enabled
Prompt > [       0.226][110580/169552 34][    0/    0  0] HASP: Loading HASP settings
Prompt > [       0.239][110580/169552 34][    0/    0  0] WIFI: Loading WiFi settings
Prompt > [       0.252][110580/169552 34][    0/    0  0] MQTT: Loading MQTT settings
Prompt > [       0.266][110580/169552 34][    0/    0  0] TELN: Loading Telnet settings
Prompt > [       0.279][110580/169552 34][    0/    0  0] MDNS: Loading MDNS settings
Prompt > [       0.292][110580/169552 34][    0/    0  0] HTTP: Loading HTTP settings
Prompt > [       0.305][110580/169552 34][    0/    0  0] GPIO: Loading GPIO settings
Prompt > [       0.318][110580/169552 34][    0/    0  0] CONF: Settings loaded
Prompt > [       0.330][110580/171100 35][    0/    0  0] MSGR: Starting...
Prompt > [       0.341][110580/171100 35][    0/    0  0] MSGR: Started
Prompt > [       0.353][110580/171100 35][    0/    0  0] TFT : Starting...
Prompt > [       0.695][110580/170948 35][    0/    0  0] TFT : Started
Prompt > [       1.061][110580/170852 35][    0/    0  0] LVGL: Starting...
Prompt > [       1.080][77812/137880 43][63044/63232  1] LVGL: LV_FS_PC is initialized with.
Prompt > [       1.102][54260/111716 51][62308/62356  1] GUI : Initialize Cursor
Prompt > [       1.117][54260/111716 51][61976/61992  1] LVGL: Started
Prompt > [       1.128][54260/111716 51][61976/61992  1] ATTR: src/hasp/hasp.cpp 544 0
Prompt > [       1.141][54260/111716 51][61976/61992  1] ATTR: src/hasp/hasp.cpp 545 0
Prompt > [       1.165][54260/111716 51][59516/60156  2] HASP: Custom theme loaded
Prompt > [       1.198][54260/111716 51][58964/58968  1] HASP: Loading /pages.jsonl
Prompt > [       1.237][54260/111716 51][58848/58852  1] HASP: Loaded /pages.jsonl
Prompt > [       1.249][54260/111716 51][58848/58852  1] HASP: Changing page to 1
Prompt > [       1.263][54260/111716 51][58652/58660  1] MQTT: Broker not configured
Prompt > [       1.275][54260/111716 51][58652/58660  1] GPIO: Starting...
Prompt > [       1.287][54260/111716 51][58652/58660  1] GPIO: Reboot counter 0
Prompt > [       1.299][54260/111716 51][58652/58660  1] GPIO: Started
Prompt > [       1.424][54260/75292 27][58652/58660  1] WIFI: Connecting to LEPWLAN
Prompt > [       1.436][54260/75404 28][58652/58660  1] TIME: TIMEZONE: EST5EDT,M3.2.0/2,M11.1.0
Prompt > [       1.450][54260/75404 28][58652/58660  1] TIME: NTPSERVER: pool.ntp.org time.nist.gov time.google.com
Prompt > [       1.469][54260/71552 24][58652/58660  1] HTTP: Started
Prompt > [       1.481][54260/70968 23][58652/58660  1] TELN: Telnet console started
Prompt > [       1.494][54260/70968 23][58652/58660  1] FTP : Starting...
Prompt > [       1.506][redacted][redacted][redacted]
Prompt > [       1.542][redacted][redacted][redacted]
Prompt >  Ftp server waiting for connection on port 21
[       3.102][redacted][redacted][redacted]
Prompt > [       3.122][redacted][redacted][redacted]
Prompt > [       4.881][redacted][redacted][redacted]
Prompt > [       4.907][redacted][redacted][redacted]
Prompt > [12:55:17.709][redacted][redacted][redacted]
Prompt > [12:55:17.741][54260/59772  9][58732/58816  1] MSGR: jsonl={"page":0,"id":239,"obj":"msgbox","text":"%ip%", "auto_close":20000}
Prompt > [12:55:17.798][redacted][redacted][redacted]
Prompt > [12:55:17.813][redacted][redacted][redacted]
Prompt > [12:55:17.827][redacted][redacted][redacted]
Prompt > [12:55:17.864][54260/59176  8][58196/58244  1] MDNS: Started
Prompt > [12:55:25.088][54260/59504  8][58144/58240  1] MQTT: Unkown error p0b239 => {"event":"down","val":0,"text":"OK"}
Prompt > [12:55:25.161][54260/59504  8][58196/58244  1] MQTT: Unkown error p0b239 => {"event":"up","val":0,"text":"OK"}
```

**Reply by fvanroie** [reply]
> Should it show something more by default?

Nope, that **is** the default layout that comes out-of-the-box.
You can now customize the layout, or upload some of the example layouts from the documentation to the `pages.jsonl` file on the device.

https://youtu.be/QgE4WrgHrWg?t=1679
